pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -2800,12 +2806,34 @@ private CompletableFuture<Void> doTransactionAcknowledgeForResponse(MessageId me
             } else {
                 bitSetRecyclable.clear(messageIdAdv.getBatchIndex());
             }
-            cmd = Commands.newAck(consumerId, ledgerId, entryId, bitSetRecyclable, ackType, validationError, properties,
-                    txnID.getLeastSigBits(), txnID.getMostSigBits(), requestId, messageIdAdv.getBatchSize());
+            cmdList.add(Commands.newAck(consumerId, ledgerId, entryId, bitSetRecyclable, ackType, validationError, properties,
+                    txnID.getLeastSigBits(), txnID.getMostSigBits(), requestId, messageIdAdv.getBatchSize()));
             bitSetRecyclable.recycle();
         } else {
-            cmd = Commands.newAck(consumerId, ledgerId, entryId, null, ackType, validationError, properties,
-                    txnID.getLeastSigBits(), txnID.getMostSigBits(), requestId);
+            MessageIdImpl[] chunkMsgIds = this.unAckedChunkedMessageIdSequenceMap.remove(messageIdAdv);
+            // cumulative ack chunk by the last messageId
+            if (chunkMsgIds == null || ackType == AckType.Cumulative) {
+                cmdList.add(Commands.newAck(consumerId, ledgerId, entryId, null, ackType, validationError,
+                        properties, txnID.getLeastSigBits(), txnID.getMostSigBits(), requestId));
```

Comment:
Same as the above comment.

Reply:
There is only one ack command for cumulative ack. Adding it to cmdList is to unify the form.